src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -230,8 +232,61 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
                     propertyStoreItems[property.Name] = item;
                 }
 
+                return CombineMetadataTypeAttributes(propertyStoreItems);
+            }
+
+            [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
```

Comment:
```suggestion
```

This attribute isn't valid. `DynamicallyAccessedMembers` are only valid to annotate a `Type` or a `string` (which is a type name). Or it can also be applied to a `class` as well.

Here you are saying the `return` value should keep `All` members, but the return value is a `Dictionary<string, PropertyStoreItem>`.

I'm surprised the linker doesn't complain about this (cc @vitek-karas @sbomer).

Reply:
Good catch @eerhardt - filed an issue in the linker: https://github.com/mono/linker/issues/2065

Reply:
Yes I had initially pasted that in while testing and forgot to check if it was actually required.